### ISSUE TITLE
spectral does not resolve path->parameters->$ref

### ISSUE THREAD

**Issue by naivefun** [stoplightio/spectral#108]
```yaml
/domains/{domainCode}:
  parameters:
    - $ref: "#/components/parameters/domainCode"
```

The domainCode is defined in `components/parameters/domainCode` but it complains:
```
- name: path-params
  summary: Path parameters are correct and valid.
  message: >-
    The path "**/domains/{domainCode}/models**" uses a parameter
    "**{domainCode}**" that does not have a corresponding definition.


    To fix, add a path parameter with the name "**domainCode**".
  path:
    - paths
    - '/domains/{domainCode}/models'
  severity: 50
  severityLabel: error
```

**Comment by tbarn** [comment]
Hey @naivefun, we've been working on a v2 of Spectral. We improved on how it handles $refs so I'll see if this happens there right now.

**Comment by philsturgeon** [comment]
Hey just taking a quick look at the example, but `/domains/{domainCode}` and `/domains/{domainCode}/models` are two different paths. Could it be that the 2nd one does not have the ref?

**Comment by naivefun** [comment]
@philsturgeon sorry the example has flaws but the point is as the title :)

**Comment by philsturgeon** [comment]
Ah! I was trying to debug based on what I had, and that looked like it could have been the issue. 

Maybe you could post more of the YAML? If you could share as little as you are comfortable sharing which still produced the error, I can run it locally and see what is going on.

**Comment by blazeworx** [comment]
Also experiencing this at the moment.

Here's an example:

spec.yaml:
```yaml
openapi: 3.0.0
info:
  title: Do Something API
  description: This API Should do something..
  version: 0.1.0
  contact:
    name: API Support
    url: 'http://do.something.com/support'
    email: do@something.com

servers:
  - url: 'http://api.something.com/do'
    description: Some Server

paths:
  '/something/{some_param}':
    get:
      summary: Yo, Do Something.
      description: |
        This endpoint does something
      operationId: doSomethingWooYeah
      tags:
        - Something
      parameters:
        - $ref: './some_param.yaml'
      responses:
        '200':
          description: Yay OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  result:
                    type: boolean
```

some_param.yaml:
```yaml
in: path
name: some_param
required: true
description: Some Param to do something else
schema:
  type: string
  enum:
    - yes
    - no
    - maybe
```

![image](https://user-images.githubusercontent.com/4718238/55434529-25affc80-55db-11e9-94b4-b7ed179fb190.png)

When I put the contents of the some_param.yaml file into the main spec file and just inline that definition then all is well. 

![image](https://user-images.githubusercontent.com/4718238/55434744-a3740800-55db-11e9-9de2-349564736c95.png)

My current work around is to use resolve all refs before running spectral lint.

**Comment by philsturgeon** [comment]
@naivefun hey there! I never heard back from you about getting a sample of the YAML to recreate this bug, but I'd wager a bet that the v2.0 release fixed it. Grab a later version and give it a try! Installation instructions on the README :)

@blazeworx I'm pretty sure you have a different issue. Before resolving was not happening _at all_ by default, but now we automatically handle internal references only. External refs are not supported, and that is what you have here: `  - $ref: './some_param.yaml'`

We plan to make multiple resolution strategies available in future versions, but for now we just do whatever https://github.com/stoplightio/json-ref-resolver does by default. Until then, the solution is to do exactly what you are already doing, process the original files however you like and pass the result into Spectral.

**Comment by blazeworx** [comment]
Thanks @philsturgeon. We have multiple (very) large API specs and we've spent a lot of time breaking them up into smaller files into a sane and logical folder/file structure so that maintaining them is easier (primarily because we do everything spec first). For now we're still using `speccy` for linting in our pipeline because it handles external refs.

As mentioned earlier, I was able to get `spectral lint` working by using `speccy resolve` and then passing that to `spectral lint` but I do not want to introduce yet another dependency and step when `speccy` does the job in one step for now. I eventually want to replace `speccy` with `spectral` because we need more flexibility for custom rules and I like the error output which is a bit more informative. So I guess until https://github.com/stoplightio/json-ref-resolver can resolve external references then we'll have to stick to what we have for now.

**Comment by philsturgeon** [comment]
I completely understand. We can close this issue as the work is already on the roadmap, and it will be sooner rather than later don’t worry.
